src/Juvix/Prelude/Path.hs

```diff
@@ -1,28 +1,91 @@
 
--- | Synonym for Path.</>. Useful to avoid name clashes
 infixr 5 <//>
 
+-- | Synonym for Path.</>. Useful to avoid name clashes
 (<//>) :: Path b Dir -> Path Rel t -> Path b t
 (<//>) = (Path.</>)
 
+someFile :: FilePath -> SomeBase File
+someFile r = fromMaybe (error ("not a file path: " <> pack r)) (parseSomeFile r)
+
+someDir :: FilePath -> SomeBase Dir
+someDir r = fromMaybe (error ("not a dir path: " <> pack r)) (parseSomeDir r)
+
 relFile :: FilePath -> Path Rel File
-relFile = fromJust . parseRelFile
+relFile r = fromMaybe (error ("not a relative file path: " <> pack r)) (parseRelFile r)
 
 relDir :: FilePath -> Path Rel Dir
-relDir = fromJust . parseRelDir
+relDir r = fromMaybe (error ("not a relative directory path: " <> pack r)) (parseRelDir r)
+
+absFile :: FilePath -> Path Abs File
+absFile r = fromMaybe (error ("not an absolute file path: " <> pack r)) (parseAbsFile r)
+
+destructAbsDir :: Path Abs Dir -> (Path Abs Dir, [Path Rel Dir])
+destructAbsDir d = go d []
+  where
+    go :: Path Abs Dir -> [Path Rel Dir] -> (Path Abs Dir, [Path Rel Dir])
+    go p acc
+      | isRoot p = (p, acc)
+      | otherwise = go (parent p) (dirname p : acc)
+
+isRoot :: Path a Dir -> Bool
+isRoot p = parent p == p
+
+-- | is the root of absolute files always "/" ?
+destructAbsFile :: Path Abs File -> (Path Abs Dir, [Path Rel Dir], Path Rel File)
+destructAbsFile x = (root, dirs, filename x)
+  where
+    (root, dirs) = destructAbsDir (parent x)
+
+isJuvixFile :: Path b File -> Bool
+isJuvixFile = (== Just ".juvix") . fileExtension
+
+isHiddenDirectory :: Path b Dir -> Bool
+isHiddenDirectory p = case toFilePath (dirname p) of
+  "./" -> False
+  '.' : _ -> True
+  _ -> False
+
+parseRelFile' :: FilePath -> Path Rel File
+parseRelFile' = fromJust . parseRelFile
+
+someBaseToAbs :: Path Abs Dir -> SomeBase b -> Path Abs b
+someBaseToAbs root = \case
+  Rel r -> root <//> r
+  Abs a -> a
+
+removeExtension :: Path b File -> Maybe (Path b File)
+removeExtension = fmap fst . splitExtension
+
+removeExtension' :: Path b File -> Path b File
+removeExtension' = fst . fromJust . splitExtension
+
+replaceExtension' :: String -> Path b File -> Path b File
+replaceExtension' ext = fromJust . replaceExtension ext
 
-destructPath :: Path b Dir -> [Path Rel Dir]
-destructPath p = map relDir (splitPath (toFilePath p))
+parents :: Path Abs a -> NonEmpty (Path Abs Dir)
+parents = go [] . parent
+  where
+    go :: [Path Abs Dir] -> Path Abs Dir -> NonEmpty (Path Abs Dir)
+    go ac p
+      | isRoot p = NonEmpty.reverse (p :| ac)
+      | otherwise = go (p : ac) (parent p)
 
-destructFilePath :: Path b File -> ([Path Rel Dir], Path Rel File)
-destructFilePath p = case nonEmptyUnsnoc (nonEmpty' (splitPath (toFilePath p))) of
-  (ps, f) -> (fmap relDir (maybe [] toList ps), relFile f)
+withTempDir' :: (MonadIO m, MonadMask m) => (Path Abs Dir -> m a) -> m a
+withTempDir' = withTempDir $(mkRelDir ".") "tmp"
```

Comment:
I think we should use [withSystemTempDir](https://hackage.haskell.org/package/path-io-1.7.0/docs/Path-IO.html#v:withSystemTempDir) instead of withTempDir to avoid adding the temporary directory to the workspace.

Reply:
done in 2a249828259abd40a5a85a10f57d5da460ea5582